data-entry/dataEntry.app.js

```diff
@@ -78,12 +78,62 @@
                         if (key.simple) {
                             (function(key) {
                                 key.getDomainValues().then(function success(values) {
-                                    dataEntryModel.domainValues[key.colset.columns[0].name] = [];
-                                    var domainValues = dataEntryModel.domainValues[key.colset.columns[0].name];
-
-                                    angular.forEach(values.data, function(value) {
-                                        var field = Object.keys(value)[0];
-                                        domainValues.push(value[field]);
+                                    var domainValues = dataEntryModel.domainValues[key.colset.columns[0].name] = [];
+                                    var table = key.key._table;
+                                    var annotations = table.annotations.all();
+                                    var termColumnName = null;
+
+
+                                    // TODO need to figure out how to handle if mutiple annotations are present
+                                    angular.forEach(annotations, function(annotation) {
+                                        // there should only be one term annotation
+                                        if (annotation.content.term) {
+                                            termColumnName = annotation.content.term;
+                                        }
+                                    });
+                                    // if a term annotation is defined
+                                    if (termColumnName) {
+                                        // do nothing and break out of "if"
+                                    }
+                                    // use the term column instead
+                                    else if(table.columns._columns.Term) {
+                                        termColumnName = "Term";
+                                    }
+                                    // next choice is name
+                                    else if(table.columns._columns.Name) {
+                                        termColumnName = "Name";
+                                    }
+                                    // find a column with unambigous text field ?
+                                    else {
+                                        // TODO
+                                        // angular.forEach(table.columns.all(), function(column) {
+                                        //
+                                        // });
+                                    };
+
+                                    var foreignKeyTablePath = new ERMrest.DataPath(table);
+                                    foreignKeyTablePath.entity.get().then(function success(entitySet) {
+                                        angular.forEach(values.data, function(value) {
```

Comment:
I find this whole `forEach` block confusing. Maybe some nuance of angular I don't understand or maybe the code could be clearer. I'd like it to be clearer. Maybe some comments in here so I can follow the logic.